src/byte_set.rs

```diff
@@ -126,21 +132,64 @@ impl<const LEN: usize> From<[u8; LEN]> for ByteSet {
     }
 }
 
-impl From<Range<u8>> for ByteSet {
-    fn from(range: Range<u8>) -> ByteSet {
+impl<'a> From<&'a [u8]> for ByteSet {
+    fn from(bytes: &'a [u8]) -> ByteSet {
         let mut bs = ByteSet::new();
-        for b in range {
-            bs.insert(b);
+        for b in bytes {
+            bs.insert(*b);
         }
         bs
     }
 }
 
-impl From<RangeInclusive<u8>> for ByteSet {
-    fn from(range: RangeInclusive<u8>) -> ByteSet {
-        let mut bs = ByteSet::new();
-        for b in range {
-            bs.insert(b);
+impl From<ops::Range<u8>> for ByteSet {
```

Comment:
Does this need qualification given that ops is already imported?

Reply:
I fix this later on in the utf-8 PR so it would be a difficult merge if I changed it at this point. But point taken.